components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -1659,6 +1765,167 @@ public Response apisHead(String query, String ifNoneMatch, MessageContext messag
+        return null;
+    }
+    /**
+     * Validate graphQL Schema
+     * @param fileInputStream  input file
+     * @param fileDetail file Detail
+     * @param messageContext messageContext
+     * @return Validation response
+     */
+    @Override
+    public Response apisValidateGraphqlSchemaPost(InputStream fileInputStream, Attachment fileDetail, MessageContext messageContext) {
+
+        String errorMessage = "";
+        String schema;
+        TypeDefinitionRegistry typeRegistry;
+        Set<SchemaValidationError> validationErrors;
+        boolean isValid = false;
+        SchemaParser schemaParser = new SchemaParser();
+        GraphQLValidationResponseDTO validationResponse = new GraphQLValidationResponseDTO();
+
+        try {
+            schema = IOUtils.toString(fileInputStream, RestApiConstants.CHARSET);
+
+            if (schema.isEmpty()) {
+                errorMessage = "GraphQL Schema cannot be empty or null to validate it";
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+
+            typeRegistry = schemaParser.parse(schema);
+            GraphQLSchema graphQLSchema = UnExecutableSchemaGenerator.makeUnExecutableSchema(typeRegistry);
+            SchemaValidator schemaValidation = new SchemaValidator();
+            validationErrors = schemaValidation.validateSchema(graphQLSchema);
+
+            if (validationErrors.toArray().length > 0) {
+                errorMessage = "InValid Schema";
+            } else {
+                isValid = true;
+                validationResponse.setIsValid(isValid);
+                GraphQLValidationResponseGraphQLInfoDTO graphQLInfo = new GraphQLValidationResponseGraphQLInfoDTO();
+                List<APIOperationsDTO> operationArray = extractGraphQLOperationList(schema);
+                graphQLInfo.setOperations(operationArray);
+                GraphQLSchemaDTO schemaObj = new GraphQLSchemaDTO();
+                schemaObj.setSchemaDefinition(schema);
+                graphQLInfo.setGraphQLSchema(schemaObj);
+                validationResponse.setGraphQLInfo(graphQLInfo);
+            }
+        } catch (SchemaProblem | IOException e) {
+            errorMessage = e.getMessage();
+        }
+
+        if(!isValid) {
+            validationResponse.setIsValid(isValid);
+            validationResponse.setErrorMessage(errorMessage);
+        }
+        return Response.ok().entity(validationResponse).build();
+    }
+
+    /**
+     * Extract GraphQL Operations from given schema
+     * @param schema graphQL Schema
+     * @return the arrayList of APIOperationsDTO
+     */
+    private List<APIOperationsDTO> extractGraphQLOperationList(String schema) {
+        List<APIOperationsDTO> operationArray = new ArrayList<>();
+        SchemaParser schemaParser = new SchemaParser();
+        TypeDefinitionRegistry typeRegistry = schemaParser.parse(schema);
+        Map<java.lang.String, graphql.language.TypeDefinition> operationList = typeRegistry.types();
```

Comment:
Why do we need java.lang part?

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80